From d8cc2dd6c05a0006e03323d78d5b0e71d0c34b4e Mon Sep 17 00:00:00 2001
Message-Id: <d8cc2dd6c05a0006e03323d78d5b0e71d0c34b4e.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: amanna <animesh.manna@intel.com>
Date: Wed, 21 Jan 2015 15:49:02 +0530
Subject: [PATCH 05/29] SQUASHME! [VPG]: drm/i915: Corrected default DSPARB

SQUASHME! - This patch should be squashed into the following existing patch:
	Author: Arun R Murthy <arun.r.murthy@intel.com>
	Change-Id: I459c46602297346418cb85c35779d3ec3a3e3071
	REVERTME [VPG]: drm/i915: Program DDL, Water marks and DSPARB as per
	atomicity

Current arbiter value divide the pipe fifo as follow-
- 50% of fifo for primary plane.
- Rest 50% equally divided between two sprites plane.

Issue: GMINL-4878
Change-Id: Ifae5ced933062b3cf943dd59e28b9d9723e6f0b7
Signed-off-by: Animesh Manna <animesh.manna@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h      |    2 +-
 drivers/gpu/drm/i915/intel_display.c |    2 +-
 drivers/gpu/drm/i915/intel_pm.c      |    2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 8ffb9f2..58684df 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -4109,7 +4109,7 @@ enum punit_power_well {
 #define   DPINVGTT_STATUS_MASK_CHV		0xfff
 
 #define	DSPARB	(dev_priv->info.display_mmio_offset + 0x70030)
-#define	DSPARB_VLV_DEFAULT	0xc080c080
+#define	DSPARB_VLV_DEFAULT	0x80008000
 #define	DSPARB_CSTART_MASK	(0x7f << 7)
 #define	DSPARB_CSTART_SHIFT	7
 #define	DSPARB_BSTART_MASK	(0x7f)
diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 10c45c7..70d9c46 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2550,6 +2550,7 @@ static void intel_find_plane_obj(struct intel_crtc *intel_crtc,
 
 static void vlv_update_watermarks(struct drm_i915_private *dev_priv)
 {
+	I915_WRITE(DSPARB, DSPARB_VLV_DEFAULT);
 	I915_WRITE(DSPFW1,
 		   (DSPFW_SR_VAL << DSPFW_SR_SHIFT) |
 		   (DSPFW_CURSORB_VAL << DSPFW_CURSORB_SHIFT) |
@@ -2575,7 +2576,6 @@ static void vlv_update_watermarks(struct drm_i915_private *dev_priv)
 			(DSPFW7_SPRITED_VAL << DSPFW7_SPRITED_SHIFT) |
 			(DSPFW7_SPRITEC1_VAL << DSPFW7_SPRITEC1_SHIFT) |
 			DSPFW7_SPRITEC_VAL);
-	I915_WRITE(DSPARB, DSPARB_VLV_DEFAULT);
 }
 
 int i915_set_plane_180_rotation(struct drm_device *dev, void *data,
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index cb886af..5ee1691 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -1598,6 +1598,7 @@ static void valleyview_update_wm(struct drm_crtc *crtc)
 			    &planeb_wm, &cursorb_wm))
 		enabled |= 1 << PIPE_B;
 
+	I915_WRITE(DSPARB, DSPARB_VLV_DEFAULT);
 	I915_WRITE(DSPFW1,
 		   (DSPFW_SR_VAL << DSPFW_SR_SHIFT) |
 		   (DSPFW_CURSORB_VAL << DSPFW_CURSORB_SHIFT) |
@@ -1615,7 +1616,6 @@ static void valleyview_update_wm(struct drm_crtc *crtc)
 			(DSPFW5_CURSORB_VAL << DSPFW5_CURSORB_SHIFT) |
 			DSPFW5_CURSORSR_VAL);
 	I915_WRITE(DSPFW6, DSPFW6_DISPLAYSR_VAL);
-	I915_WRITE(DSPARB, DSPARB_VLV_DEFAULT);
 }
 
 static void g4x_update_wm(struct drm_crtc *crtc)
-- 
1.7.9.5

